CtrlVAF/CtrlVAF.Core/Core/LicensedDispatcher.cs

```diff
@@ -0,0 +1,171 @@
+﻿using CtrlVAF.Core.Attributes;
+using CtrlVAF.Models;
+
+using MFiles.VAF.Configuration;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace CtrlVAF.Core
+{
+    /// <summary>
+    /// Decorator for a <see cref="Dispatcher{TReturn}"/> that checks the application's licensing status.
+    /// Works with <see cref="LicenseRequiredAttribute"/>
+    /// </summary>
+    /// <typeparam name="TReturn">The return type of the dispatcher</typeparam>
+    public class LicensedDispatcher<TReturn> : Dispatcher<TReturn>
+    {
+        private LicenseContentBase license;
+        private Dispatcher<TReturn> dispatcher;
+
+        /// <summary>
+        /// Wraps a dispatcher with logic checking the application's licensing status when evaluating types.
+        /// </summary>
+        /// <param name="dispatcher">the wrapped dispatcher</param>
+        /// <param name="license">the license contents for the VaultApplication</param>
+        public LicensedDispatcher(Dispatcher<TReturn> dispatcher, LicenseContentBase license = null)
+        {
+            this.license = license;
+            this.dispatcher = dispatcher;
+        }
+
+        public override TReturn Dispatch(params ICtrlVAFCommand[] commands)
+        {
+            dispatcher.IncludeAssemblies(Assembly.GetCallingAssembly());
+
+            var types =  GetTypes(commands);
+
+            if (!types.Any())
+                return default;
+
+            return HandleConcreteTypes(types, commands);
+        }
+
+        protected internal override IEnumerable<Type> GetTypes(params ICtrlVAFCommand[] commands)
```

Comment:
This method to me looks like exactly the same as the other GetTypes method in the alternative LicensedDispatcher class below.
I would probably consider including this method in a LicensedDispatcher_Common abstract class and have both LicensedDispatcher inherit from this.

Reply:
I can't implement them commonly because their return types are different. At least that's what I think. 
This way I thought it was cleaner to have them seperately instead of void Dispatch() methods needing an object return type.

Reply:
The return type of the GetTypes method is the same both here and on line 118. The code is exactly the same

Reply:
nvm I thought you meant the HandleConcreteTypes method for some reason. The gettypes is exactly the same I think yeah. I'll extract that then.

Reply:
Ah the issue is that LicensedDispatcher already inherits from Dispatcher and you can't inherit from two classes at once. 

I created a static helper class to implement the functionality of the filter.